mongodbatlas/resource_mongodbatlas_org_invitation.go

```diff
@@ -0,0 +1,244 @@
+package mongodbatlas
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	matlas "go.mongodb.org/atlas/mongodbatlas"
+)
+
+func resourceMongoDBAtlasOrgInvitation() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceMongoDBAtlasOrgInvitationCreate,
+		ReadContext:   resourceMongoDBAtlasOrgInvitationRead,
+		DeleteContext: resourceMongoDBAtlasOrgInvitationDelete,
+		UpdateContext: resourceMongoDBAtlasOrgInvitationUpdate,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceMongoDBAtlasOrgInvitationImportState,
+		},
+		Schema: map[string]*schema.Schema{
+			"org_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"username": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"invitation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expires_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"roles": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+					ValidateFunc: validation.StringInSlice([]string{
+						"ORG_OWNER",
+						"ORG_GROUP_CREATOR",
+						"ORG_BILLING_ADMIN",
+						"ORG_READ_ONLY",
+						"ORG_MEMBER",
+					}, false),
+				},
+			},
+		},
+	}
+}
+
+func resourceMongoDBAtlasOrgInvitationRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// Get client connection.
+	conn := meta.(*MongoDBClient).Atlas
+	ids := decodeStateID(d.Id())
+	orgID := ids["org_id"]
+	username := ids["username"]
+	invitationID := ids["invitation_id"]
+
+	orgInvitation, _, err := conn.Organizations.Invitation(ctx, orgID, invitationID)
+	if err != nil {
+		// case 404
+		// deleted in the backend case
+		var target *matlas.ErrorResponse
+		if errors.As(err, &target) && target.ErrorCode == "NOT_FOUND" {
+			d.SetId("")
+			return nil
+		}
```

Comment:
FYI @coderGo93 and @thetonymaster this is a more modern go way to check errors (go1.13) it would be good to stick to this type of error checking from now on